### PR TITLE
NECO-163: can get each metadata

### PR DESCRIPTION
**Pull request by YukiThornton**
## Ticket
- https://pflabo.atlassian.net/browse/NECO-163
## Changes
- Added another meta method(mediaId, fieldName)
- Added a private method replaceUserMeta
## Tests
- See the ticket.

### CODE REVIEW

File: Source/MediaStorage.swift [hunk 1]
```diff
@@ -364,4 +371,94 @@ public class MediaStorage {
...
+            regex = try NSRegularExpression(pattern: pattern, options: NSRegularExpressionOptions.AllowCommentsAndWhitespace)
+            let matches = regex.matchesInString(userMeta, options: NSMatchingOptions.ReportProgress, range:NSMakeRange(0, userMeta.characters.count))
+            if matches.count > 0 {
+                return userMeta.substringFromIndex(userMeta.startIndex.advancedBy(5))
+            } else {
+                return ""
+            }
+        } catch {
+            return ""
+        }
```
Comment (sohda) [inline Source/MediaStorage.swift]: I think following flow is simpler. How about you?

``` swift
let replace = "$1"
let replaceString = userMeta.stringByReplacingOccurrencesOfString(replaceUserMetaRegex, withString: replace, options: NSStringCompareOptions.RegularExpressionSearch, range: nil)

return replaceString == userMeta ? "" : replaceString
```

I tried the above code in Playground.

Comment (YukiThornton) [inline Source/MediaStorage.swift]: Thank you! It looks simpler than my code. I'll replace it with this codes.

File: Source/MediaStorage.swift [hunk 2]
```diff
@@ -364,4 +372,84 @@ public class MediaStorage {
...
+    }
+
+    private func replaceUserMeta (userMeta: String!) -> String{
+        let replacedString = userMeta.stringByReplacingOccurrencesOfString(replaceUserMetaRegex, withString: firstGroupRegex, options: NSStringCompareOptions.RegularExpressionSearch, range: nil)
+
+        return replacedString == userMeta ? "" : replacedString
+
+    }
+
```
Comment (sohda) [inline Source/MediaStorage.swift:454]: Is this blank line necessary ?

File: Source/MediaStorage.swift [hunk 3]
```diff
@@ -364,4 +372,84 @@ public class MediaStorage {
...
+            )
+
+        }
+    }
+
+    private func replaceUserMeta (userMeta: String!) -> String{
+        let replacedString = userMeta.stringByReplacingOccurrencesOfString(replaceUserMetaRegex, withString: firstGroupRegex, options: NSStringCompareOptions.RegularExpressionSearch, range: nil)
+
+        return replacedString == userMeta ? "" : replacedString
+
```
Comment (sohda) [inline Source/MediaStorage.swift:452]: Is this blank line necessary ?

File: Source/MediaStorage.swift [hunk 4]
```diff
@@ -364,4 +372,84 @@ public class MediaStorage {
...
+                    )
+                }
+
+            }
+
+            completionHandler(
+                dataDic,
+                MediaStorageError(statusCode: nil, message: nil)
+            )
+
```
Comment (sohda) [inline Source/MediaStorage.swift:444]: Is this blank line necessary ?

File: Source/MediaStorage.swift [hunk 5]
```diff
@@ -364,4 +372,84 @@ public class MediaStorage {
...
+            } else  {
+                do {
+                    dataDic = try NSJSONSerialization.JSONObjectWithData(data!, options: NSJSONReadingOptions.AllowFragments) as! [String : String]
+                } catch {
+                    completionHandler(
+                        Dictionary<String, String>(),
+                        MediaStorageError(statusCode: statusCode, message: "invalid response: \(dataString)")
+                    )
+                }
+
```
Comment (sohda) [inline Source/MediaStorage.swift:437]: Is this blank line necessary ?

Comment (YukiThornton) [inline Source/MediaStorage.swift:437]: According to [swift style guide](https://github.com/jarinosuke/swift-style-guide), you can "Make liberal use of vertical whitespace to divide code into logical chunks." So, I used a lot of blank lines to divide codes into chunks.
Those lines are not necessary but it does not affect how this SDK works. And I found other blank lines like [here](https://github.com/hishida/media-storage-swift/pull/14/files#diff-67514e2fb02036161bb26f1a6ed338fdR288). So I think that it's unnecessary to delete those lines now.

File: Source/MediaStorage.swift (hunk 4 above)
Comment (YukiThornton) [inline Source/MediaStorage.swift:444]: Same as this [comment](https://github.com/hishida/media-storage-swift/pull/14/files#r70571589).

File: Source/MediaStorage.swift (hunk 3 above)
Comment (YukiThornton) [inline Source/MediaStorage.swift:452]: Same as this [comment](https://github.com/hishida/media-storage-swift/pull/14/files#r70571589).

File: Source/MediaStorage.swift (hunk 2 above)
Comment (YukiThornton) [inline Source/MediaStorage.swift:454]: Same as this [comment](https://github.com/hishida/media-storage-swift/pull/14/files#r70571589).

File: Source/MediaStorage.swift (hunk 5 above)
Comment (sohda) [inline Source/MediaStorage.swift:437]: Okay. I'm going to merge this pull request.
I know the style guide. I meant some method (like `connect` , `download`)  are not inserted whitespace at end of them.

**Pull request merged**